Gems/ROS2/Code/Source/RobotImporter/FixURDF/FixURDF.cpp

```diff
@@ -0,0 +1,136 @@
+ *
+ */
+
+#include "FixURDF.h"
+#include <AzCore/XML/rapidxml_print.h>
+#include <AzCore/std/containers/unordered_set.h>
+#include <iostream>
+
+namespace ROS2::Utils
+{
+    AZStd::vector<AZStd::string> AddMissingInertiaToLinks(AZ::rapidxml::xml_node<>* urdf)
+    {
+        AZStd::vector<AZStd::string> modifiedLinks;
+        using namespace AZ::rapidxml;
+        AZStd::vector<xml_node<>*> links;
+        for (xml_node<>* link = urdf->first_node("link"); link; link = link->next_sibling("link"))
+        {
+            if (!link->first_node("inertial"))
+            {
+                links.push_back(link);
+            }
+        }
+
+        for (auto* link : links)
+        {
+            xml_node<>* inertial = urdf->document()->allocate_node(node_element, "inertial");
+
+            xml_node<>* mass = urdf->document()->allocate_node(node_element, "mass");
+            mass->append_attribute(urdf->document()->allocate_attribute("value", "1."));
+            inertial->append_node(mass);
+            xml_node<>* inertia = urdf->document()->allocate_node(node_element, "inertia");
+
+            inertia->append_attribute(urdf->document()->allocate_attribute("ixx", "1."));
+            inertia->append_attribute(urdf->document()->allocate_attribute("ixy", "0."));
+            inertia->append_attribute(urdf->document()->allocate_attribute("ixz", "0."));
+
+            inertia->append_attribute(urdf->document()->allocate_attribute("iyx", "0."));
+            inertia->append_attribute(urdf->document()->allocate_attribute("iyy", "1."));
+            inertia->append_attribute(urdf->document()->allocate_attribute("iyz", "0."));
+
+            inertia->append_attribute(urdf->document()->allocate_attribute("izx", "0."));
+            inertia->append_attribute(urdf->document()->allocate_attribute("izy", "0."));
+            inertia->append_attribute(urdf->document()->allocate_attribute("izz", "1."));
+
+            inertial->append_node(inertia);
+
+            xml_node<>* origin = urdf->document()->allocate_node(node_element, "origin");
+            origin->append_attribute(urdf->document()->allocate_attribute("xyz", "0. 0. 0."));
+            inertial->append_node(origin);
+
+            auto* name = link->first_attribute("name");
+            if (name)
+            {
+                modifiedLinks.push_back(name->value());
+            }
+            link->append_node(inertial);
+        }
+
+        return modifiedLinks;
+    }
+
+    AZStd::vector<AZStd::string> ChangeDuplications(AZ::rapidxml::xml_node<>* urdf)
+    {
+        using namespace AZ::rapidxml;
+        AZStd::vector<AZStd::string> modifiedLinks;
+        AZStd::unordered_set<AZStd::string> linkAndJointsName;
+        for (xml_node<>* link = urdf->first_node("link"); link; link = link->next_sibling("link"))
+        {
+            auto* name = link->first_attribute("name");
+            if (name)
+            {
+                linkAndJointsName.insert(name->value());
+            }
```

Comment:
Does this need to check for duplicate link names too, or is that deliberately ignored here to let the SDF converter throw the error? If so, maybe add a comment here to explain that, something like "// Duplicate link names are invalid in a URDF file, so they won't be handled here. The SDF converter will detect the error case and throw an error."

Reply:
Done, change the name of the function to `RenameDuplicatedJoints`